core/src/main/java/com/avast/metrics/core/multi/GroupMonitor.java

```diff
@@ -0,0 +1,150 @@
+package com.avast.metrics.core.multi;
+
+import com.avast.metrics.TimerPairImpl;
+import com.avast.metrics.api.*;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Wrapper around a group of monitors.
+ */
+public class GroupMonitor implements Monitor {
+
+    private final List<Monitor> monitors;
+    private final Naming naming;
+
+    /**
+     * Factory method.
+     *
+     * @param monitors   list of monitors
+     * @return multiple  monitor containing all passed monitors
+     */
+    public static Monitor of(Monitor... monitors) {
+        return new GroupMonitor(Arrays.asList(monitors),Naming.defaultNaming());
+    }
+
+    /**
+     * Factory method.
+     *
+     * @param monitors   list of monitors
+     * @param naming     naming conventions for TimerPairs
+     * @return multiple  monitor containing all passed monitors
+     */
+    public static Monitor of(Naming naming, Monitor... monitors) {
```

Comment:
What about to hardcode the naming here and use the default? This feature is over-engineered.
cc @jakubjanecek

Reply:
I do agree that `Naming` is over engineered, but it's already there I can't do much about it

Reply:
you can probably get rid of it - virtually noone is using it

Reply:
JJ made to add it there, but already said many times that he changed his mind and wants it gone

Reply:
@tomasherman Why he didn't remove it before major release then? :man_facepalming: